ingest/expression_files/dense_ingestor.py

```diff
@@ -0,0 +1,187 @@
+"""Module for ingesting dense matrix files
+
+DESCRIPTION
+Module provides extract and transforms function for gene expression data for
+an dense matrix.
+
+"""
+import collections
+import csv
+import datetime
+import sys
+from typing import List  # noqa: F401
+
+from bson.objectid import ObjectId
+
+try:
+    from expression_files import GeneExpression
+    sys.path.append("../ingest")
+    from ingest_files import IngestFiles
+
+except ImportError:
+    # Used when importing as external package, e.g. imports in single_cell_portal code
+    from .expression_files import GeneExpression
+    sys.path.append("../ingest")
+    from ..ingest_files import IngestFiles
+
+
+class DenseIngestor(GeneExpression, IngestFiles):
+    ALLOWED_FILE_TYPES = ["text/csv",
+                          "text/plain", "text/tab-separated-values"]
+
+    def __init__(self, file_path, study_file_id, study_id, **kwargs):
+        GeneExpression.__init__(self, file_path, study_file_id, study_id)
+        IngestFiles.__init__(
+            self, file_path, allowed_file_types=self.ALLOWED_FILE_TYPES
+        )
+        self.matrix_params = kwargs
+        self.csv_file_handler, self.file_handler = self.open_file(
+            self.file_path)
+        self.gene_names = {}
+        self.header = next(self.csv_file_handler)
```

Comment:
Unless you have reason otherwise, I would move this statement to inside the execute_ingest method, since reading the header seems like more of an ingest-y thing to do

Reply:
aab53bd